### PR TITLE
[TIR] Allow missing TensorIntrin in registry lookup

### PR DESCRIPTION
**Pull request by vinx13**
Added an option to allow missing tensor intrin.

cc @junrushao @spectrometerHBH

### CODE REVIEW

**Comment by junrushao** [comment]
Would you like to provide some context BTW? like, in which cases do we allow it missing?

File: include/tvm/tir/function.h [hunk 1]
```diff
@@ -233,10 +233,13 @@ class TensorIntrin : public ObjectRef {
...
    * \param name The name of the TensorIntrin.
+   * \param allow_missing Whether to allow missing tensor intrin. If false, an exception is raised
+   *    if the tensor intrin is not found.
    * \return The TensorIntrin with the specified name.
-   * \throws This method throws an exception if the TensorIntrin does not exist.
+   * \throws This method throws an exception if the TensorIntrin does not exist and allow_missing is
+   * false.
    */
-  TVM_DLL static TensorIntrin Get(String name);
+  TVM_DLL static TensorIntrin Get(String name, bool allow_missing = false);
```
Comment (junrushao) [inline include/tvm/tir/function.h:242]:
```suggestion
  TVM_DLL static Optional<TensorIntrin> Get(String name, bool allow_missing = false);
```

File: python/tvm/tir/function.py [hunk 2]
```diff
@@ -228,20 +228,24 @@ def register(name: str, desc: PrimFunc, impl: PrimFunc):
...
         name : str
             The name of the TensorIntrin to look up.
 
+        allow_missing : bool
+            Whether to allow missing tensor intrin. If False, raise an error if the tensor intrin
+        doesn't exist.
+
         Returns
         -------
         result : TensorIntrin
```
Comment (junrushao) [inline python/tvm/tir/function.py:245]: use Optional

**Comment by vinx13** [comment]
One use case is check if tensor interns exists, it will allow further automation like generate tensor intrins if missing

**Comment by junrushao** [comment]
Got it! Can we add a simple test where the intrin name is missing?

**Review by junrushao — APPROVED** [review]

**Pull request merged**